source/Server/Ldap/NestedGroupFinder.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Octopus.Server.Extensibility.Authentication.Ldap.Model;
+
+namespace Octopus.Server.Extensibility.Authentication.Ldap
+{
+    public interface INestedGroupFinder
+    {
+        IEnumerable<GroupDistinguishedName> FindAllParentGroups(LdapContext context, int depth, IEnumerable<GroupDistinguishedName> names);
+    }
+
+    public class NestedGroupFinder : INestedGroupFinder
+    {
+        readonly IGroupParentFinder parentFinder;
+
+        public NestedGroupFinder(IGroupParentFinder parentFinder)
+        {
+            this.parentFinder = parentFinder;
+        }
+
+        public IEnumerable<GroupDistinguishedName> FindAllParentGroups(LdapContext context, int searchDepth, IEnumerable<GroupDistinguishedName> names)
+        {
+            var groups = new HashSet<GroupDistinguishedName>(names.Distinct());
+            var nextGroups = groups.ToList();
```

Comment:
Curious why `ToList` here?

Reply:
Good pick up - I have no idea 🤷‍♂️ 
Fixed it now.